code/load_db.py

```diff
@@ -0,0 +1,63 @@
+import pyodbc
+import teardown as td
+import extract_transform as et
+import config_handler as db_
+
+
+# import pyodbc
+# from configparser import ConfigParser
+
+
+def main():
+    # For rerun purposes (incase of new daily feed)
+    td.db_cleanup()
+
+    # Extract Json and transform
+    dataframe = et.json_to_df()
+
+    # write to MS SQL db
+    conn = db_.ms_sql_connection()
+    cursor = conn.cursor()
+    # Create Tables
+    cursor.execute("""CREATE TABLE [Top_rated_Movie] (
+                   Id int NOT NULL CONSTRAINT [Id] PRIMARY KEY,
+                   title char (100) NULL ,
+                   [year] [date] NULL ,
+                   [genres] [nvarchar] (50) NULL ,
+                   [duration] [int] NULL ,
+                   [releaseDate] [nvarchar] (50) NULL ,
+                   [actors] [nvarchar] (500) NULL ,
+                   [imdbRating] [varchar] (50) NULL ,
+                   ) ON [PRIMARY] """)
+
+    cursor.execute("""CREATE TABLE [Movie_Actor_Relationship] (
+                   	[movieID] [int] NOT NULL , 
+                   	[actor] [nvarchar] (500) NULL ,
+                   	[imdbRating] [float] (50) NULL ,) 
+                   """)
+
+    cursor.execute("""
+                    CREATE TABLE [Movie_Genre_Relationship] (
+                   	[movieID] [int] NOT NULL , 
+                   	[genre] [nvarchar] (50) NULL ,
+                   	[imdbRating] [float] (50) NULL ,
+                   )""")
+    conn.commit()
+
+    # Inserting data in SQL Table:-
+    for index, row in dataframe.iterrows():
+        cursor.execute(
```

Comment:
I guess here you compile SQL-query on each iteration. Maybe it's cached internally, but typically it makes sense to create pre-compiled queries.

Reply:
updated